model/histogram/generic.go

```diff
@@ -325,3 +326,186 @@ func compactBuckets[IBC internalBucketCount](buckets []IBC, spans []Span, maxEmp
 
 	return buckets, spans
 }
+
+func getBound(idx, schema int32) float64 {
```

Comment:
Where do the numbers 1024 1025 and frac==0.5 come from? I feel it is not trivial to get my head around the math here.

Reply:
Yeah, it requires a lecture about IEEE 754 floats, essentially. I'll try my best…